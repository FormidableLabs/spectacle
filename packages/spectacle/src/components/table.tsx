import styled from 'styled-components';
import {
  color,
  typography,
  space,
  compose,
  border,
  layout,
  ColorProps,
  TypographyProps,
  SpaceProps,
  BorderProps,
  LayoutProps
} from 'styled-system';

export type TableProps = ColorProps &
  TypographyProps &
  SpaceProps &
  BorderProps &
  LayoutProps;

const Table = styled.table.attrs<TableProps>((props) => ({
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin',
  width: 1,
  ...props
}))<TableProps>(compose(color, typography, space, border, layout));

const TableHeader = styled.thead.attrs<TableProps>((props) => ({
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  fontWeight: 'bold',
  textAlign: 'left',
  margin: 'listMargin',
  ...props
}))<TableProps>(compose(color, typography, space, border, layout));

const TableBody = styled.tbody.attrs<TableProps>((props) => ({
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin',
  width: 1,
  ...props
}))<TableProps>(compose(color, typography, space, border, layout));

const TableRow = styled.tr.attrs<TableProps>((props) => ({
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin',
  ...props
}))<TableProps>(compose(color, typography, space, border, layout));

const TableCell = styled.td.attrs<TableProps>((props) => ({
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin',
  ...props
}))<TableProps>(compose(color, typography, space, border, layout));

export { Table, TableCell, TableRow, TableHeader, TableBody };
