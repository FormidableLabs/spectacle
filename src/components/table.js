import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getStyles } from '../utils/base';
import styled from 'react-emotion';

const StyledTable = styled.table(props => props.styles);

export default class Table extends Component {
  render() {
    return (
      <StyledTable
        className={this.props.className}
        styles={[
          this.context.styles.components.table,
          getStyles.call(this),
          this.props.style
        ]}
      >
        {this.props.children}
      </StyledTable>
    );
  }
}

Table.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object
};

Table.contextTypes = {
  styles: PropTypes.object,
  store: PropTypes.object
};
