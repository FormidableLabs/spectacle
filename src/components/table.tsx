import styled from 'styled-components';
import {
  color,
  typography,
  space,
  compose,
  border,
  layout,
  ColorProps,
  TypographyProps,
  SpaceProps,
  BorderProps,
  LayoutProps
} from 'styled-system';

export type TableProps = ColorProps &
  TypographyProps &
  SpaceProps &
  BorderProps &
  LayoutProps;

const Table = styled.table<TableProps>(
  compose(color, typography, space, border, layout)
);

Table.defaultProps = {
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin',
  width: 1
};

const TableHeader = styled.thead<TableProps>(
  compose(color, typography, space, border, layout)
);

TableHeader.defaultProps = {
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  fontWeight: 'bold',
  textAlign: 'left',
  margin: 'listMargin'
};

const TableBody = styled.tbody<TableProps>(
  compose(color, typography, space, border, layout)
);

TableBody.defaultProps = {
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin',
  width: 1
};

const TableRow = styled.tr<TableProps>(
  compose(color, typography, space, border, layout)
);

TableRow.defaultProps = {
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin'
};

const TableCell = styled.td<TableProps>(
  compose(color, typography, space, border, layout)
);

TableCell.defaultProps = {
  color: 'primary',
  fontFamily: 'text',
  fontSize: 'text',
  textAlign: 'left',
  margin: 'listMargin'
};

export { Table, TableCell, TableRow, TableHeader, TableBody };
