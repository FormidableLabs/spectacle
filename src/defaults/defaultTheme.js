const defaultTheme = {};

export default defaultTheme;
