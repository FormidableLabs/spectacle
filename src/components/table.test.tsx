/* eslint-disable react/prop-types */
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ThemeProvider } from 'styled-components';

import defaultTheme from '../theme/default-theme';
import { Table, TableRow, TableCell, TableBody, TableHeader } from './table';

Enzyme.configure({ adapter: new Adapter() });

const mountWithTheme = tree => {
  const WrappingThemeProvider = props => (
    <ThemeProvider theme={defaultTheme}>{props.children}</ThemeProvider>
  );
  return mount(tree, { wrappingComponent: WrappingThemeProvider });
};

describe('<Table />', () => {
  it('should render a <table> with a <tr> for each row and a <td> with text for each cell', () => {
    const wrapper = mountWithTheme(
      <Table>
        <TableBody>
          <TableRow>
            <TableCell>Row 1</TableCell>
            <TableCell>Row 1</TableCell>
          </TableRow>
          <TableRow>
            <TableCell>Row 2</TableCell>
            <TableCell>Row 2</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <table> with bold text for the header row', () => {
    const wrapper = mountWithTheme(
      <Table>
        <TableHeader>
          <TableRow>
            <TableCell>Row 1</TableCell>
            <TableCell>Row 1</TableCell>
          </TableRow>
        </TableHeader>
      </Table>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
