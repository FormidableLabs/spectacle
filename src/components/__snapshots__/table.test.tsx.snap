// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render a <table> with a <tr> for each row and a <td> with text for each cell 1`] = `
<styled.table
  color="primary"
  fontFamily="text"
  fontSize="text"
  margin="listMargin"
  textAlign="left"
  width={1}
>
  <StyledComponent
    color="primary"
    fontFamily="text"
    fontSize="text"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {
          "color": "primary",
          "fontFamily": "text",
          "fontSize": "text",
          "margin": "listMargin",
          "textAlign": "left",
          "width": 1,
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "gdxKB",
          "rules": Array [
            [Function],
          ],
        },
        "displayName": "styled.table",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "table",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    margin="listMargin"
    textAlign="left"
    width={1}
  >
    <table
      className="sc-bdVaJa gdxKB"
      color="primary"
      fontFamily="text"
      fontSize="text"
      width={1}
    >
      <styled.tbody
        color="primary"
        fontFamily="text"
        fontSize="text"
        margin="listMargin"
        textAlign="left"
        width={1}
      >
        <StyledComponent
          color="primary"
          fontFamily="text"
          fontSize="text"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "_foldedDefaultProps": Object {
                "color": "primary",
                "fontFamily": "text",
                "fontSize": "text",
                "margin": "listMargin",
                "textAlign": "left",
                "width": 1,
              },
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": false,
                "lastClassName": "jvKwEj",
                "rules": Array [
                  [Function],
                ],
              },
              "displayName": "styled.tbody",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "tbody",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          margin="listMargin"
          textAlign="left"
          width={1}
        >
          <tbody
            className="sc-htpNat jvKwEj"
            color="primary"
            fontFamily="text"
            fontSize="text"
            width={1}
          >
            <styled.tr
              color="primary"
              fontFamily="text"
              fontSize="text"
              margin="listMargin"
              textAlign="left"
            >
              <StyledComponent
                color="primary"
                fontFamily="text"
                fontSize="text"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "_foldedDefaultProps": Object {
                      "color": "primary",
                      "fontFamily": "text",
                      "fontSize": "text",
                      "margin": "listMargin",
                      "textAlign": "left",
                    },
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bxivhb",
                      "isStatic": false,
                      "lastClassName": "bmEQpW",
                      "rules": Array [
                        [Function],
                      ],
                    },
                    "displayName": "styled.tr",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bxivhb",
                    "target": "tr",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                margin="listMargin"
                textAlign="left"
              >
                <tr
                  className="sc-bxivhb bmEQpW"
                  color="primary"
                  fontFamily="text"
                  fontSize="text"
                >
                  <styled.td
                    color="primary"
                    fontFamily="text"
                    fontSize="text"
                    margin="listMargin"
                    textAlign="left"
                  >
                    <StyledComponent
                      color="primary"
                      fontFamily="text"
                      fontSize="text"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "color": "primary",
                            "fontFamily": "text",
                            "fontSize": "text",
                            "margin": "listMargin",
                            "textAlign": "left",
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "ebewS",
                            "rules": Array [
                              [Function],
                            ],
                          },
                          "displayName": "styled.td",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "td",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      margin="listMargin"
                      textAlign="left"
                    >
                      <td
                        className="sc-ifAKCX ebewS"
                        color="primary"
                        fontFamily="text"
                        fontSize="text"
                      >
                        Row 1
                      </td>
                    </StyledComponent>
                  </styled.td>
                  <styled.td
                    color="primary"
                    fontFamily="text"
                    fontSize="text"
                    margin="listMargin"
                    textAlign="left"
                  >
                    <StyledComponent
                      color="primary"
                      fontFamily="text"
                      fontSize="text"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "color": "primary",
                            "fontFamily": "text",
                            "fontSize": "text",
                            "margin": "listMargin",
                            "textAlign": "left",
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "ebewS",
                            "rules": Array [
                              [Function],
                            ],
                          },
                          "displayName": "styled.td",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "td",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      margin="listMargin"
                      textAlign="left"
                    >
                      <td
                        className="sc-ifAKCX ebewS"
                        color="primary"
                        fontFamily="text"
                        fontSize="text"
                      >
                        Row 1
                      </td>
                    </StyledComponent>
                  </styled.td>
                </tr>
              </StyledComponent>
            </styled.tr>
            <styled.tr
              color="primary"
              fontFamily="text"
              fontSize="text"
              margin="listMargin"
              textAlign="left"
            >
              <StyledComponent
                color="primary"
                fontFamily="text"
                fontSize="text"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "_foldedDefaultProps": Object {
                      "color": "primary",
                      "fontFamily": "text",
                      "fontSize": "text",
                      "margin": "listMargin",
                      "textAlign": "left",
                    },
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bxivhb",
                      "isStatic": false,
                      "lastClassName": "bmEQpW",
                      "rules": Array [
                        [Function],
                      ],
                    },
                    "displayName": "styled.tr",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bxivhb",
                    "target": "tr",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                margin="listMargin"
                textAlign="left"
              >
                <tr
                  className="sc-bxivhb bmEQpW"
                  color="primary"
                  fontFamily="text"
                  fontSize="text"
                >
                  <styled.td
                    color="primary"
                    fontFamily="text"
                    fontSize="text"
                    margin="listMargin"
                    textAlign="left"
                  >
                    <StyledComponent
                      color="primary"
                      fontFamily="text"
                      fontSize="text"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "color": "primary",
                            "fontFamily": "text",
                            "fontSize": "text",
                            "margin": "listMargin",
                            "textAlign": "left",
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "ebewS",
                            "rules": Array [
                              [Function],
                            ],
                          },
                          "displayName": "styled.td",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "td",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      margin="listMargin"
                      textAlign="left"
                    >
                      <td
                        className="sc-ifAKCX ebewS"
                        color="primary"
                        fontFamily="text"
                        fontSize="text"
                      >
                        Row 2
                      </td>
                    </StyledComponent>
                  </styled.td>
                  <styled.td
                    color="primary"
                    fontFamily="text"
                    fontSize="text"
                    margin="listMargin"
                    textAlign="left"
                  >
                    <StyledComponent
                      color="primary"
                      fontFamily="text"
                      fontSize="text"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "color": "primary",
                            "fontFamily": "text",
                            "fontSize": "text",
                            "margin": "listMargin",
                            "textAlign": "left",
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "ebewS",
                            "rules": Array [
                              [Function],
                            ],
                          },
                          "displayName": "styled.td",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "td",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      margin="listMargin"
                      textAlign="left"
                    >
                      <td
                        className="sc-ifAKCX ebewS"
                        color="primary"
                        fontFamily="text"
                        fontSize="text"
                      >
                        Row 2
                      </td>
                    </StyledComponent>
                  </styled.td>
                </tr>
              </StyledComponent>
            </styled.tr>
          </tbody>
        </StyledComponent>
      </styled.tbody>
    </table>
  </StyledComponent>
</styled.table>
`;

exports[`<Table /> should render a <table> with bold text for the header row 1`] = `
<styled.table
  color="primary"
  fontFamily="text"
  fontSize="text"
  margin="listMargin"
  textAlign="left"
  width={1}
>
  <StyledComponent
    color="primary"
    fontFamily="text"
    fontSize="text"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "_foldedDefaultProps": Object {
          "color": "primary",
          "fontFamily": "text",
          "fontSize": "text",
          "margin": "listMargin",
          "textAlign": "left",
          "width": 1,
        },
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "gdxKB",
          "rules": Array [
            [Function],
          ],
        },
        "displayName": "styled.table",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "table",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    margin="listMargin"
    textAlign="left"
    width={1}
  >
    <table
      className="sc-bdVaJa gdxKB"
      color="primary"
      fontFamily="text"
      fontSize="text"
      width={1}
    >
      <styled.thead
        color="primary"
        fontFamily="text"
        fontSize="text"
        fontWeight="bold"
        margin="listMargin"
        textAlign="left"
      >
        <StyledComponent
          color="primary"
          fontFamily="text"
          fontSize="text"
          fontWeight="bold"
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "_foldedDefaultProps": Object {
                "color": "primary",
                "fontFamily": "text",
                "fontSize": "text",
                "fontWeight": "bold",
                "margin": "listMargin",
                "textAlign": "left",
              },
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": false,
                "lastClassName": "biMVCL",
                "rules": Array [
                  [Function],
                ],
              },
              "displayName": "styled.thead",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "thead",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          margin="listMargin"
          textAlign="left"
        >
          <thead
            className="sc-bwzfXH biMVCL"
            color="primary"
            fontFamily="text"
            fontSize="text"
            fontWeight="bold"
          >
            <styled.tr
              color="primary"
              fontFamily="text"
              fontSize="text"
              margin="listMargin"
              textAlign="left"
            >
              <StyledComponent
                color="primary"
                fontFamily="text"
                fontSize="text"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "_foldedDefaultProps": Object {
                      "color": "primary",
                      "fontFamily": "text",
                      "fontSize": "text",
                      "margin": "listMargin",
                      "textAlign": "left",
                    },
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bxivhb",
                      "isStatic": false,
                      "lastClassName": "bmEQpW",
                      "rules": Array [
                        [Function],
                      ],
                    },
                    "displayName": "styled.tr",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bxivhb",
                    "target": "tr",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                margin="listMargin"
                textAlign="left"
              >
                <tr
                  className="sc-bxivhb bmEQpW"
                  color="primary"
                  fontFamily="text"
                  fontSize="text"
                >
                  <styled.td
                    color="primary"
                    fontFamily="text"
                    fontSize="text"
                    margin="listMargin"
                    textAlign="left"
                  >
                    <StyledComponent
                      color="primary"
                      fontFamily="text"
                      fontSize="text"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "color": "primary",
                            "fontFamily": "text",
                            "fontSize": "text",
                            "margin": "listMargin",
                            "textAlign": "left",
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "ebewS",
                            "rules": Array [
                              [Function],
                            ],
                          },
                          "displayName": "styled.td",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "td",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      margin="listMargin"
                      textAlign="left"
                    >
                      <td
                        className="sc-ifAKCX ebewS"
                        color="primary"
                        fontFamily="text"
                        fontSize="text"
                      >
                        Row 1
                      </td>
                    </StyledComponent>
                  </styled.td>
                  <styled.td
                    color="primary"
                    fontFamily="text"
                    fontSize="text"
                    margin="listMargin"
                    textAlign="left"
                  >
                    <StyledComponent
                      color="primary"
                      fontFamily="text"
                      fontSize="text"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "color": "primary",
                            "fontFamily": "text",
                            "fontSize": "text",
                            "margin": "listMargin",
                            "textAlign": "left",
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": false,
                            "lastClassName": "ebewS",
                            "rules": Array [
                              [Function],
                            ],
                          },
                          "displayName": "styled.td",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "td",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      margin="listMargin"
                      textAlign="left"
                    >
                      <td
                        className="sc-ifAKCX ebewS"
                        color="primary"
                        fontFamily="text"
                        fontSize="text"
                      >
                        Row 1
                      </td>
                    </StyledComponent>
                  </styled.td>
                </tr>
              </StyledComponent>
            </styled.tr>
          </thead>
        </StyledComponent>
      </styled.thead>
    </table>
  </StyledComponent>
</styled.table>
`;
